cassiopeia/preprocess/UMI_utils.py

```diff
@@ -215,20 +231,28 @@ def form_collapsed_clusters(
 
     # Read in the AlignmentFile again as iterating over it in the previous for
     # loop has destructively removed all alignments from the file object
-    sorted_als = pysam.AlignmentFile(str(sorted_fn), check_sq=False)
+    sorted_als = pysam.AlignmentFile(sorted_fn, check_sq=False)
 
     sorted_als = progress(sorted_als, total=total_reads, desc="Collapsing UMIs")
 
     cell_groups = utilities.group_by(sorted_als, cell_key)
 
     with pysam.AlignmentFile(
-        str(collapsed_fn), "wb", header=empty_header
+        collapsed_fn, "wb", header=empty_header
     ) as collapsed_fh:
         for cell_BC, cell_group in cell_groups:
             for UMI, UMI_group in utilities.group_by(cell_group, UMI_key):
-                clusters = form_clusters(
-                    UMI_group, max_read_length, max_hq_mismatches
-                )
+                if method == "cutoff":
+                    clusters = form_clusters(
+                        UMI_group, max_read_length, max_hq_mismatches
+                    )
+                elif method == "bayesian":
+                    clusters = form_clusters_bayesian(
+                        UMI_group,
+                        proportion=max_hq_mismatches / max_read_length,
+                    )
+                else:
+                    raise UMIUtilsError(f"Unknown method `{method}`")
```

Comment:
Let's make this error message a bit more specific -- something like `Unknown method in forming UMI clusters: {method}`